### PR TITLE
Me/dpc 4119 track long lived jobs

### PR DESCRIPTION
**Pull request by MEspositoE14s**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4119

## 🛠 Changes

- Updated `AwsDistributedBatchQueue` so that in addition to emitting a queue size metric to AWS, it also logs queue age for Splunk to pick up an alert on.  Queue age is defined as the age in hours of the oldest job in queued status.
- Also updated so that 0 queue size metrics are now emitted to AWS.  Before, 0's weren't sent.

## ℹ️ Context

In the past, we've had jobs get stuck in the queue without ever knowing about it.  By writing the queue age to the logs, a Splunk alert can be configured that will let us know if a job gets stuck in the queue for too long.

Splunk alert can be found [here](https://splunk.cloud.cms.gov/en-US/app/Office_of_Enterprise_Data_Analytics/alert?s=%2FservicesNS%2Fnobody%2FOffice_of_Enterprise_Data_Analytics%2Fsaved%2Fsearches%2FDPC%2520Prod%2520-%2520Long%2520Lived%2520Jobs).  The alert currently looks for jobs that have been queued for more than 20 hours, but that will probably have to be adjusted once we're done making performance improvements to `dpc-aggregation` and we have a better idea what our baseline looks like.

Run book for handling an alert can be found [here](https://confluence.cms.gov/display/DAPC/Splunk+Alerts%3A+Long+Lived+Jobs).

## 🧪 Validation

- Ran locally and made sure logs were written with the correct queue age.
- Deployed to dev and made sure Splunk alert was triggered.
Screenshot 2024-07-02 at 4 19 04 PM

### CODE REVIEW

File: dpc-queue/src/main/java/gov/cms/dpc/queue/JobQueueModule.java [hunk 1]
```diff
@@ -79,43 +80,50 @@ UUID provideAggregatorID() {
...
-            return provideSlf4jReporter(metricRegistry);
-        }
+    private CloudWatchReporter provideCloudWatchReporter(MetricRegistry metricRegistry, String metricName, CloudWatchAsyncClient cloudWatchAsyncClient) {
+        return CloudWatchReporter.forRegistry(
+                metricRegistry,
+                cloudWatchAsyncClient,
+                configuration().getDpcAwsQueueConfiguration().getAwsNamespace()
+            )
+            .withReportRawCountValue()
+            .withZeroValuesSubmission()
```
Comment (MEspositoE14s) [inline dpc-queue/src/main/java/gov/cms/dpc/queue/JobQueueModule.java:118]: Not necessarily part of this ticket, but this line tells the reporter to emit 0 values to AWS.

File: dpc-aggregation/src/main/resources/application.yml [hunk 2]
```diff
@@ -89,8 +90,10 @@ bbclient:
 awsQueue:
   emitAwsMetrics: ${EMIT_AWS_METRICS:-true}
   queueSizeMetricName: ${QUEUE_SIZE_METRIC_NAME:-"JobQueueBatchCount"}
+  queueAgeMetricName: ${QUEUE_AGE_METRIC_NAME:-"JobQueueBatchAgeHours"}
   awsRegion: ${AWS_REGION:-"us-east-1"}
-  awsReportingInterval: ${AWS_REPORTING_INTERVAL:-60}
+  awsSizeReportingInterval: ${AWS_SIZE_REPORTING_INTERVAL:-60}
+  awsAgeReportingInterval: ${AWS_AGE_REPORTING_INTERVAL:-600}
```
Comment (MEspositoE14s) [inline dpc-aggregation/src/main/resources/application.yml:96]: We're reporting queue size every minute, and queue age every ten minutes.  I took a guess at what I thought these values should be, but am definitely open to changing them.

Comment (jdettmannnava) [reply dpc-aggregation/src/main/resources/application.yml:96]: These defaults make sense to me. For the size reporting interval, if we want more reactivity, we can decrease the interval with environment variables anyway.

**Review by jdettmannnava — APPROVED** [review]
LGTM

**Pull request merged**